test/core/storage/trie/polkadot_trie/polkadot_trie_cursor_test.cpp

```diff
@@ -132,32 +134,107 @@ TEST_F(PolkadotTrieCursorTest, BigPseudoRandomTrieRandomStart) {
-      {"06070801"_hex2buf, "06070801"_hex2buf},
-      {"06070802"_hex2buf, "06070802"_hex2buf},
-      {"06070803"_hex2buf, "06070803"_hex2buf}};
-  auto trie = makeTrie(vals);
+  auto trie = makeTrie(lex_sorted_vals);
   auto c = trie->cursor();
   EXPECT_OUTCOME_FALSE_1(c->seek("f"_buf));
   EXPECT_OUTCOME_TRUE_1(c->seek("06"_hex2buf));
   Buffer prev_key{0};
   do {
-    EXPECT_OUTCOME_TRUE(key, c->key());
+    ASSERT_TRUE(c->key());
+    auto key = c->key().value();
     ASSERT_LT(prev_key, key);
     prev_key = key;
     EXPECT_OUTCOME_TRUE_1(c->next());
   } while (c->isValid());
 }
+
+/**
+ * @given a non-empty trie
+ * @when seeking a lower bound for a given byte sequence which is somewhere in
+ * the middle in the set of trie keys and is not present in the trie
+ * @then the corresponding lower bound is found
+ */
+TEST_F(PolkadotTrieCursorTest, LowerBoundKeyNotPresent) {
+  auto trie = makeTrie(lex_sorted_vals);
+  auto c = trie->trieCursor();
+  c->seekLowerBound("06066666"_hex2buf).value();
+  ASSERT_EQ(c->value().value(), "0607"_hex2buf);
+  EXPECT_OUTCOME_TRUE_1(c->next());
+  ASSERT_EQ(c->value().value(), "060708"_hex2buf);
+}
+
+/**
+ * @given a non-empty trie
+ * @when seeking a lower bound for a given byte sequence which is somewhere in
+ * the middle in the set of trie keys
+ * @then the corresponding lower bound is found
+ */
+TEST_F(PolkadotTrieCursorTest, LowerBoundMiddleFromRoot) {
+  auto trie = makeTrie(lex_sorted_vals);
+  auto c = trie->trieCursor();
+  c->seekLowerBound("03"_hex2buf).value();
+  ASSERT_EQ(c->value().value(), "05"_hex2buf);
+  EXPECT_OUTCOME_TRUE_1(c->next());
+  ASSERT_EQ(c->value().value(), "06"_hex2buf);
+}
+
+/**
+ * @given a non-empty trie
+ * @when seeking a lower bound for a byte sequence lex. smallest than any key in
+ * the trie
+ * @then the first key is found
+ */
+TEST_F(PolkadotTrieCursorTest, LowerBoundFirstKey) {
+  auto trie = makeTrie(lex_sorted_vals);
+  auto c = trie->trieCursor();
+
+  c->seekLowerBound("00"_hex2buf).value();
+  ASSERT_EQ(c->value().value(), "0102"_hex2buf);
+  EXPECT_OUTCOME_TRUE_1(c->next());
+  ASSERT_EQ(c->value().value(), "0103"_hex2buf);
+}
+
+/**
+ * @given an empty trie
+ * @when seeking a lower bound for a byte sequence in it
+ * @then resulting iterator is not valid
+ */
+TEST_F(PolkadotTrieCursorTest, LowerBoundEmptyTrie) {
+  auto trie = makeTrie({});
+  auto c = trie->trieCursor();
+
+  EXPECT_OUTCOME_TRUE_1(c->seekLowerBound("00"_hex2buf));
+  ASSERT_FALSE(c->key().has_value());
+}
+
+/**
+ * @given a non-empty trie
+ * @when seeking a lower bound in the trie for a given byte sequence
+ * @then the returned key is none if there is no lex. greater key in the trie
+ */
+TEST_F(PolkadotTrieCursorTest, LexOrderKept) {
+  auto trie =
+      makeTrie({{":heappages"_buf, "00"_hex2buf}, {":code"_buf, "geass"_buf}});
+  auto c = trie->trieCursor();
+
+  EXPECT_OUTCOME_TRUE_1(c->seekLowerBound("Optional"_buf));
+  EXPECT_FALSE(c->key().has_value());
+}
```

Comment:
Maybe makes sense to add test for loweBound of key that is bigger than all keys in a trie